tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -164,6 +189,33 @@ public function testErrorsWithPresenceRequired()
     }
 
     /**
+     * Test that errors() can work with nested data.
+     *
+     * @return void
+     */
+    public function testErrorsWithNestedFields()
+    {
+        $validator = new Validator;
+        $validator->add('user.username', 'letter', ['rule' => 'alphanumeric']);
+        $validator->add('comments.0.comment', 'letter', ['rule' => 'alphanumeric']);
```

Comment:
I don't have a good answer for that right now. One option would be to use a limited set of `Hash` path expressions. That would be better than making another path expansion syntax variant.